controllers/studentController.ts

```diff
@@ -14,15 +14,15 @@ const mailValidator = [
 const sendStudentMail = async (req: Request, res: Response): Promise<void> => {
   validation.checkErrors(req);
 
-  const { email } = req.body;
+  const { email, source } = req.body;
   await sendMail1(email);
 
   res.json({
     // eslint-disable-next-line max-len
     message: 'Nous vous avons envoyé un mail avec toutes les informations sur le dispositif. Pensez à verifier vos spams et n\'hesitez pas à nous contacter en cas de problèmes',
   });
 
-  dbStudents.insert(email);
+  dbStudents.insert(email, source);
```

Comment:
you have to add the source to the `mailValidator`and possibly sanitize it as its directly insert in the db